Content.Server/Medical/CryoPodSystem.cs

```diff
@@ -0,0 +1,359 @@
+    [Dependency] private readonly MobStateSystem _mobStateSystem = default!;
+
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<CryoPodComponent, ComponentInit>(OnComponentInit);
+        SubscribeLocalEvent<CryoPodComponent, AtmosDeviceUpdateEvent>(OnCryoPodUpdateAtmosphere);
+        SubscribeLocalEvent<CryoPodComponent, GetVerbsEvent<AlternativeVerb>>(AddAlternativeVerbs);
+        SubscribeLocalEvent<CryoPodComponent, DragDropEvent>(HandleDragDropOn);
+        SubscribeLocalEvent<CryoPodComponent, InteractHandEvent>(HandleInteractHand);
+        SubscribeLocalEvent<CryoPodComponent, InteractUsingEvent>(OnInteractUsing);
+        SubscribeLocalEvent<DoInsertCryoPodEvent>(DoInsertCryoPod);
+        SubscribeLocalEvent<CryoPodComponent, ExaminedEvent>(OnExamined);
+        SubscribeLocalEvent<CryoPodComponent, PowerChangedEvent>(OnPowerChanged);
+        SubscribeLocalEvent<CryoPodComponent, GotEmaggedEvent>(OnEmagged);
+        SubscribeLocalEvent<CryoPodPryFinished>(OnCryoPodPryFinished);
+        SubscribeLocalEvent<CryoPodPryInterrupted>(OnCryoPodPryInterrupted);
+        SubscribeLocalEvent<CryoPodComponent, DestructionEventArgs>(OnDestroyed);
+    }
+
+    private void OnComponentInit(EntityUid uid, CryoPodComponent cryoPodComponent, ComponentInit args)
+    {
+        base.Initialize();
+        cryoPodComponent.BodyContainer = _containerSystem.EnsureContainer<ContainerSlot>(uid, $"scanner-bodyContainer");
+    }
+
+    private void UpdateAppearance(EntityUid uid, CryoPodComponent? cryoPod = null)
+    {
+        if (!Resolve(uid, ref cryoPod))
+            return;
+
+        _appearanceSystem.SetData(uid, SharedCryoPodComponent.CryoPodVisuals.IsOpen, cryoPod.BodyContainer.ContainedEntity == null);
+        _appearanceSystem.SetData(uid, SharedCryoPodComponent.CryoPodVisuals.IsOn, cryoPod.Enabled);
+        if (TryComp<PointLightComponent>(uid, out var light))
+        {
+            light.Enabled = cryoPod.Enabled && cryoPod.BodyContainer.ContainedEntity != null;
+        }
+
+        _appearanceSystem.SetData(uid,SharedCryoPodComponent.CryoPodVisuals.PanelOpen, false);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+        foreach (var cryoPod in EntityQuery<CryoPodComponent>())
+        {
+            if (!cryoPod.Enabled)
+            {
+                continue;
+            }
+            cryoPod.Accumulator += frameTime;
+
+            if (cryoPod.Accumulator < cryoPod.BeakerTransferTime)
+                continue;
+
+            cryoPod.Accumulator -= cryoPod.BeakerTransferTime;
+
+            var container = _itemSlotsSystem.GetItem(cryoPod.Owner, "beakerSlot");
+            if (container != null
+                && container.Value.Valid
+                && cryoPod.BodyContainer.ContainedEntity != null
+                && TryComp<BloodstreamComponent>(cryoPod.BodyContainer.ContainedEntity, out var bloodstream)
```

Comment:
```suggestion
                && TryComp<BloodstreamComponent>(cryoPod.BodyContainer.ContainedEntity, out var bloodstream)
```

Should also be using entityqueries.